yt_dlp/extractor/instagram.py

```diff
@@ -630,16 +648,13 @@ class InstagramStoryIE(InstagramBaseIE):
     def _real_extract(self, url):
         username, story_id = self._match_valid_url(url).groups()
 
-        story_info_url = f'{username}/{story_id}/?__a=1' if username == 'highlights' else f'{username}/?__a=1'
-        story_info = self._download_json(f'https://www.instagram.com/stories/{story_info_url}', story_id, headers={
-            'X-IG-App-ID': 936619743392459,
-            'X-ASBD-ID': 198387,
-            'X-IG-WWW-Claim': 0,
-            'X-Requested-With': 'XMLHttpRequest',
-            'Referer': url,
-        })
-        user_id = story_info['user']['id']
-        highlight_title = traverse_obj(story_info, ('highlight', 'title'))
+        story_info, wh = self._download_webpage_handle(url, story_id)
```

Comment:
For storyIE, can you add a test that doesn't need login?

Reply:
ok, in that case pls confirm that story extraction is working as expected after my changes

Reply:
and whether https://github.com/yt-dlp/yt-dlp/issues/3002 is fixed

Reply:
confirming they indeed work as expected and #3002 seemed to me like a login requirement problem but its all fixed. But still need to make small cleanups, please do any if you have some.

Reply:
Done making changes, please check it and now do the other things left.